### PR TITLE
Download specific ngrok version

### PR DESCRIPTION
**Pull request by Mitalie**
Ngrok 3.0 configuration format is not backwards compatible. Rather than try to detect and support both, or force all users to update, we can keep using 2.x versions as long as ngrok servers accept them.

### CODE REVIEW

**Comment by panicstevenson** [comment]
Is this still necessary after https://github.com/OverlayPlugin/OverlayPlugin/pull/18? Sorry, I'm unsure what configs are now incompatible.

**Comment by Mitalie** [comment]
Yes.

OverlayPlugin generates a config file for ngrok and launches it with a command line parameter pointing to that config file.  Previously an user on discord reported failure to autodownload ngrok, was instructed to download manually, and reported that ngrok 3.x was complaining about unknown flag 'c' when called with `-config`. I tested that `--config` didn't break for existing users with ngrok 2.x and made a PR, but neglected to test if 3.x needed any further changes.

Once that was merged in this fork, another user reported that ngrok 3.x was complaining about invalid config file: they now require a version field. Ngrok 3.x can upgrade old configs, but only with a separate command. I read ngrok docs and found that one of the options used by OverlayPlugin has also changed, therefore it's impossible to support both ngrok 2.x and ngrok 3.x without detecting which one we're working with. I figured that staying with 2.x for now would be the least disruptive fix.

**Comment by panicstevenson** [comment]
Wow, thanks for the detailed explanation! That's very interesting. And yea, I agree with your evaluation -- it feels like if there's not a backwards compatible path forward, then we should stick with what we know (so long as it's supported 👀).

File: OverlayPlugin.Core/Controls/WSConfigPanel.cs [hunk 1]
```diff
@@ -594,30 +593,16 @@ private bool FetchNgrok(string ngrokPath)
...
-                var match = Regex.Match(dlPage, " href=\"(https://bin.equinox.io/c/[^/]+/ngrok-v3-stable-windows-" + arch + "\\.zip)\"");
-                if (match == Match.Empty)
-                {
-                    simpLogBox.AppendText("Failed to find version on the download page! Please notify ngld or some other dev working on OverlayPlugin.\r\n");
-                    return false;
-                }
+                // Use latest known 2.x because 3.x is incompatible
+                // URLs from https://dl.equinox.io/ngrok/ngrok/stable/archive
+                var ngrokUrl = Environment.Is64BitOperatingSystem ?
+                    "https://bin.equinox.io/a/8exBtGpBr59/ngrok-2.3.40-windows-amd64.zip" :
```
Comment (panicstevenson) [inline OverlayPlugin.Core/Controls/WSConfigPanel.cs:599]: How worried are we about minor patches changing the version string and the hash?

Comment (Mitalie) [reply OverlayPlugin.Core/Controls/WSConfigPanel.cs:599]: The archive page contains all previous versions too, so the link to 2.3.40 is unlikely to become unavailable. Development seems focused on the 3.x branch, so I'm not expecting new 2.x versions to appear either, but I could change it to search the archive page for the first `-windows-amd64.zip` link if you think we should.

Comment (Mitalie) [reply OverlayPlugin.Core/Controls/WSConfigPanel.cs:599]: That might break for users again if 3.x versions are ever posted on the same archive page though - so far they haven't been.

Comment (panicstevenson) [reply OverlayPlugin.Core/Controls/WSConfigPanel.cs:599]: Gotcha, looks like the archive download page has everything too, so unless 2.3.40 gets removed for a security vulnerability or something it looks like we're okay. Thanks!

**Review by panicstevenson — APPROVED** [review]

**Pull request merged**